### PR TITLE
feat: add CLI options to enable auto-updating studios

### PR DESCRIPTION
**Pull request by ricokahler**
### Description

Adds the `--enable-auto-updates` and `unstable_enableAutoUpdates` flags to the CLI build options and config.

Additionally this PR:
- Adds `importMap`  support to `renderDocument` and `sanityBuildEntries` via `node-html-parser` (see tests)
- Refactors the external array to derive the external values from the specified import map.

Note that this will only affect the built version of the Studio and local versions will not use import maps. This decision was made because we did not want to provide a react dev build however this could potentially be possible now with import maps.

### What to review

- Review the logic, did I miss anything?
- Take a look at the tests, ensure the import map is being added correctly.

### Testing

I added tests and I tried this out in the test-studio locally.

### Notes for release

N/A

### CODE REVIEW

File: packages/@sanity/cli/src/types.ts [hunk 1]
```diff
@@ -310,6 +310,8 @@ export interface CliConfig {
   graphql?: GraphQLAPIConfig[]
 
   vite?: UserViteConfig
+
+  enableAutoUpdates?: boolean
```
Comment (binoy14) [inline packages/@sanity/cli/src/types.ts:314]: Should this be `unstable_enableAutoUpdates` ? We don't want this being used at the moment

**Review by rexxars — APPROVED** [review]
Looks good! Would just add the `@internal` annotation :)

**Comment by binoy14** [comment]
* **#6707** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/6707?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6514** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/6514?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `next`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ricokahler and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/sanity/src/_internal/cli/actions/build/buildAction.ts [hunk 2]
```diff
@@ -105,6 +117,26 @@ export default async function buildSanityStudio(
 
   const trace = telemetry.trace(BuildTrace)
   trace.start()
+
+  let importMap
+
+  if (autoUpdatesEnabled) {
+    const version = encodeURIComponent(`^${installedSanityVersion}`)
+    const autoUpdatesImports = {
+      'sanity': `https://api.sanity.work/v1/modules/sanity/default/${version}`,
```
Comment (binoy14) [inline packages/sanity/src/_internal/cli/actions/build/buildAction.ts:126]: note this is changing in a PR in in the stack above

**Review by binoy14 — APPROVED** [review]
makes sense to me, I have tested it in canaries and locally

**Pull request merged**